onmt/inputters/text_dataset.py

```diff
@@ -133,14 +71,17 @@ def make_examples(text_iter, truncate, side):
             values are more or less the result of tokenizing with those
             fields.
         """
-        # this function basically reimplements what torchtext Fields and
-        # examples give you for free. It should not exist.
-        for i, line in enumerate(text_iter):
-            line = line.strip().split()
```

Comment:
This split() ensured that tokens were handled correctly in the extract_text_features call a few lines down.
Removing it ends up in considering every character as a token, so basically breaking things.
Was there a specific reason for removing it?

Reply:
First, good catch. In this particular case, I moved the `split` to the `_read_file` method and forgot to account for whether the iterator output would be tokenized. I can change it around again so values from an iterator also get split like that.

Second, I am refactoring toward the models being completely flexible with regard to tokenization. Hard-coding `split` into the database code makes it impossible to do that. It also causes the `torchtext.data.Field`'s own tokenizer not to be used, which is counterintuitive if you're familiar with torchtext, nonobvious from looking at the code in OpenNMT (I initially tried specifying the `tokenize` argument when creating the fields and was confused that it appeared not to do anything), and not really documented anywhere. In a different branch with a few commits on top of this, I've been able remove not just the splitting but also `extract_text_features`.